modules/reactor/src/meshgenerators/ControlDrumMeshGenerator.C

```diff
@@ -0,0 +1,509 @@
+  params.addRangeCheckedParam<unsigned int>(
+      "drum_intervals", 1, "drum_intervals>0", "Number of mesh intervals in drum region");
+  params.addRangeCheckedParam<Real>("pad_start_angle",
+                                    "pad_start_angle>=0 & pad_start_angle < 360",
+                                    "Starting angle of drum pad region");
+  params.addRangeCheckedParam<Real>(
+      "pad_end_angle", "pad_end_angle>0 & pad_end_angle < 720", "Ending angle of drum pad region");
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "num_azimuthal_sectors",
+      "num_azimuthal_sectors>2",
+      "Number of azimuthal sectors to sub-divide the drum region into");
+  params.addParam<std::vector<std::vector<subdomain_id_type>>>(
+      "region_ids",
+      "IDs for each radial and axial zone for assignment of region_id extra element "
+      "id. "
+      "Inner indexing is radial zones (drum inner/drum/drum outer), outer indexing is axial");
+  params.addParam<std::vector<std::vector<std::string>>>(
+      "block_names",
+      "Block names for each radial and axial zone. "
+      "Inner indexing is radial zones (drum inner/drum/drum outer), outer indexing is axial");
+
+  params.addParamNamesToGroup("region_ids assembly_type", "ID assigment");
+  params.addParamNamesToGroup("drum_inner_radius drum_outer_radius drum_inner_intervals "
+                              "drum_intervals num_azimuthal_sectors",
+                              "Drum specifications");
+  params.addParamNamesToGroup("pad_start_angle pad_end_angle", "Control pad specifications");
+
+  params.addClassDescription(
+      "This ControlDrumMeshGenerator object is designed to generate "
+      "drum-like structures, with IDs, from a reactor geometry. "
+      "These structures can be used directly within CoreMeshGenerator to stitch"
+      "control drums into a core lattice alongside AssemblyMeshGenerator structures");
+  // depletion id generation params are added
+  addDepletionIDParams(params);
+
+  return params;
+}
+
+ControlDrumMeshGenerator::ControlDrumMeshGenerator(const InputParameters & parameters)
+  : ReactorGeometryMeshBuilderBase(parameters),
+    _assembly_type(getParam<subdomain_id_type>("assembly_type")),
+    _drum_inner_radius(getParam<Real>("drum_inner_radius")),
+    _drum_outer_radius(getParam<Real>("drum_outer_radius")),
+    _extrude(getParam<bool>("extrude")),
+    _region_ids(isParamValid("region_ids")
+                    ? getParam<std::vector<std::vector<subdomain_id_type>>>("region_ids")
+                    : std::vector<std::vector<subdomain_id_type>>())
+{
+  // Initialize ReactorMeshParams object
+  initializeReactorMeshParams(getParam<MeshGeneratorName>("reactor_params"));
+
+  // Flexible stitching needs to be invoked in order to create control drum mesh
+  if (!getReactorParam<bool>(RGMB::flexible_assembly_stitching))
+    mooseError("'flexible_assembly_stitching' needs to be set to true in ReactorMeshParams in "
```

Comment:
A lot of error messages in this file are not covered. Please add test cases to increase the coverage. https://mooseframework.inl.gov/docs/PRs/28458/coverage_diff/reactor/#modules/reactor/src/meshgenerators/ControlDrumMeshGenerator.C